src/compiler/rust.rs

```diff
@@ -237,13 +238,49 @@ where
         })
         .await?;
 
-    parsed.map(move |files| {
-        trace!(
+    parsed.map(move |mut files| {
+        // HACK: Ideally, if we're compiling a Cargo package, we should be
+        // compiling it with the same files included in the published .crate
+        // package to ensure maximum compatibility. While it's possible for
+        // the crate developers to mark files as required for the compilation
+        // via mechanisms such as `include_bytes!`, this is also a hack and
+        // may leave an undesired footprint in the compilation outputs.
+        // An upstream mechanism to only track required additional files is
+        // pending at https://github.com/rust-lang/rust/pull/84029.
+        // Until then, unconditionally include Cargo.toml manifest to provide
+        // a minimal support and to keep some crates compiling that may
+        // read additional info directly from the manifest file.
+        let cargo_manifest_dir = env_vars
+            .iter()
+            .find(|(key, _)| key.to_str() == Some("CARGO_MANIFEST_DIR"))
+            .map(|(_, value)| Path::new(value));
+        const KNOWN_AUX_DEP_FILES: &[(Option<&str>, &str)] = &[
+            (None, "Cargo.toml"),
+            (Some("libp2p_wasm_ext"), "src/websockets.js"),
```

Comment:
Are those part of the `includes = []` set in the manifest? But on the other handside there might be other rather large files as well :monocle_face: - I guess it's good as is.
We might want to consider to allow easy modification via env var?

Reply:
It has to be included in the packaged and uploaded `.crate` file to crates.io - otherwise, regular builds wouldn't be possible. The `libp2p-wasm-ext` has no `include`/`exclude`, so by default it includes everything in directory IIRC.
Here's a list of files in the package:
```
xanewok@faerun-dev:~/repos/rust-libp2p/transports/wasm-ext (master)$ cargo package --list
.cargo_vcs_info.json
CHANGELOG.md
Cargo.toml
Cargo.toml.orig
src/lib.rs
src/websockets.js
```

>  We might want to consider to allow easy modification via env var?

 Do you mean unconditionally including Cargo.toml for every package?
 We have to include `src/websockets.js` for `libp2p-wasm-ext`, otherwise we can't ever build it via `sccache-dist`.

Reply:
> It has to be included in the packaged and uploaded `.crate` file to crates.io - otherwise, regular builds wouldn't be possible. The `libp2p-wasm-ext` has no `include`/`exclude`, so by default it includes everything in directory IIRC.
> Here's a list of files in the package:
> 
> ```
> xanewok@faerun-dev:~/repos/rust-libp2p/transports/wasm-ext (master)$ cargo package --list
> .cargo_vcs_info.json
> CHANGELOG.md
> Cargo.toml
> Cargo.toml.orig
> src/lib.rs
> src/websockets.js
> ```

I think that status quo of inclusion logic is good as is :)

> > We might want to consider to allow easy modification via env var?
> 
> Do you mean unconditionally including Cargo.toml for every package?
> We have to include `src/websockets.js` for `libp2p-wasm-ext`, otherwise we can't ever build it via `sccache-dist`.

I meant more on the extensability front/generalization i.e. what if another crate pops up in the need of file? We want to have rather quick turnaround, so a `CACHEPOT_EXTRA=$crate:$resource_path,..` might be something for a follow up PR (\#idea) and I have a feeling that even with the deps tracking this might be a quick fix before it's going to be adopted in the ecosystem.

Reply:
Oh, that - yeah, that's definitely a good idea. I was thinking of introducing it in a configuration file but env var sounds easy/quicker for now :+1: